moveit_ros/moveit_servo/test/servo_calcs_unit_tests.cpp

```diff
@@ -123,6 +128,43 @@ TEST_F(EnforceLimitsTests, LowJointVelocityDeltaTest)
   checkVelocityLimits(joint_model_group_, eigen_velocity);
 }
 
+TEST_F(ServoCalcsUnitTests, SingularityScaling)
+{
+  // If we are at a singularity, we should halt
+  Eigen::VectorXd commanded_twist(6);
+  commanded_twist << 1, 0, 0, 0, 0, 0;
+  // This singular Jacobian only produces motion for Joint 1
+  Eigen::MatrixXd jacobian(7, 6);
+  // clang-format off
+  jacobian <<
+    1, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0;
+  // clang-format on
+  Eigen::JacobiSVD<Eigen::MatrixXd> svd =
+      Eigen::JacobiSVD<Eigen::MatrixXd>(jacobian, Eigen::ComputeThinU | Eigen::ComputeThinV);
+  Eigen::MatrixXd matrix_s = svd.singularValues().asDiagonal();
+  Eigen::MatrixXd pseudo_inverse = svd.matrixV() * matrix_s.inverse() * svd.matrixU().transpose();
+
+  double hard_stop_singularity_threshold = 50;
+  double lower_singularity_threshold = 10;
+  double leaving_singularity_threshold_multiplier = 2;
+
+  rclcpp::Clock clock;
+  std::shared_ptr<moveit::core::RobotState> robot_state = std::make_shared<moveit::core::RobotState>(robot_model_);
+  moveit_servo::StatusCode status;
+
+  double scaling_factor = moveit_servo::velocityScalingFactorForSingularity(
+      joint_model_group_, commanded_twist, svd, pseudo_inverse, hard_stop_singularity_threshold,
+      lower_singularity_threshold, leaving_singularity_threshold_multiplier, clock, robot_state, status);
+
+  EXPECT_EQ(scaling_factor, 0);
```

Comment:
suggest using `EXPECT_DOUBLE_EQ` when comparing floating point values

reference: https://google.github.io/googletest/reference/assertions.html#EXPECT_DOUBLE_EQ

Reply:
I expect exactly zero here, i.e. all bits should be equal to zero.